expression/builtin_time.go

```diff
@@ -2585,6 +2585,42 @@ func (du *baseDateArithmitical) getIntervalFromInt(ctx sessionctx.Context, args
 	return strconv.FormatInt(interval, 10), false, nil
 }
 
+var daysByMonth = [12]int{31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31}
+
+func getMaxDayInMonth(year, month int) int {
```

Comment:
this function duplicates the functionality of `builtinLastDaySig.evalTime()`, can we extract a common function and make them share the same code logic?

Reply:
how about I extract getMaxDayInMonth to types/MysqlTime ,just like IsLeapYear?
@zz-jason

Reply:
OK. we can discuss the function name later.

Reply:
how about MaxDayInMonth?
I have another problem, `year, month, day` in MysqlTime is private, and I could not instance it in builtin_time.go ,now we only use it in types/Time ,could I add a func to instance MyqlTime , or I could only use MysqlTime by Time ?

Reply:
@zz-jason

Reply:
I think we only need "year" and "month" to get the last day of that month, so we can abstract a function like `GetLastDay()`:
```go
func getLastDay(year, month int) {
	...
}
```

And in `builtinLastDaySig.evalTime()`, after we have get the input time, we can further get the year and month info, then call this function to get the last day.

In `getFixDays()` we can call this function as well in order to get the last day.